bootloader/prebootloader/src/rtl872x/part1/sleep_handler.cpp

```diff
@@ -324,10 +326,15 @@ void sleepProcess(void) {
              */
             auto config = sleepConfigShadow.config();
             if (config->mode == HAL_SLEEP_MODE_HIBERNATE) {
+                SysTick->CTRL &= ~SysTick_CTRL_ENABLE_Msk;
+                taskENTER_CRITICAL();
                 configureDeepSleepWakeupSource(config);
                 enterDeepSleep();
                 // It should not reach here
             } else {
+                SysTick->CTRL &= ~SysTick_CTRL_ENABLE_Msk;
```

Comment:
This disables systick, which should stop the freertos scheduler from swapping in other tasks? So effectively we stop FreeRTOS at this point? Hibernate will restart, so we dont care about starting the OS again?

Reply:
Yup. Exactly right.